### ISSUE TITLE
Accessibility: When reading a page, text after a link isn't spken by a screen reader.

### ISSUE THREAD

**Issue by prater-devin** [silverbulletmd/silverbullet#294]
Silverbullet is the most accessible (with a screen reader) personal knowledge bank I've ever used. This will make my job and life a lot easier; thank you very much for creating it. There is one accessibility issue that I've found. If you paste this line into a page:

```
This is a test of [Silverbullet](https://www.silverbullet.md). Hopefully I can read this afterwards.
```

Press Enter, then arrow up, the screen reader, NVDA in my case, only reads up to the end of the link text, and does not read the second sentence. If that can be fixed, I think this will be perfect for me. Thanks so much for this.

**Comment by prater-devin** [comment]
Further notes:

* Works best with MacOS and the VoiceOver screen reader.
* Using NVDA screen reader on Windows, I cannot read text after a link, bold text, or italic text.
* On NVDA with Windows, links to other pages aren't read unless the cursor is on the text itself. Then, it isn't shown as a link.
* Doesn't work well at all with VoiceOver on an iPhone. If I read by line, it eventually stops reading and only allows an overview, not editing.

**Comment by zefhemel** [comment]
Oh wow, I know exactly nothing about screen readers. I'd have to learn about how this all works. Any help here would be appreciated.

**Comment by prater-devin** [comment]
Screen readers work by gathering info from accessibility API's, in this
case, the DOM from the web browser. I'll paste links to info that may help:


   - ARIA accessibility: Details of using AIRA to help create accessible
   web apps
   <https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA>
   - Ten tips for creating an accessible rich text editor: Some points may
   not apply, since most text is written in Markdown, but it will matter when
   users read their written text.
   <https://jkrsp.com/accessibility-for-rich-text-editors/>

One thing that may help a lot with this is to be able to turn off or on the
live preview, so that the Markdown doesn't turn into rich text unless I
turn it on. That way, text in the text area stays constant, and I can just
pop open a preview to the side or something, or in a new tab. I think that
would fix just about every issue I have.

**Comment by tristansokol** [comment]
@prater-devin do you primarily use silver bullet on desktop (web) or mobile? Would a command that you toggle markdown views be useful? 

I have also noticed that adding in `<!--` to the page breaks the markdown rendering after it so that might be useful for you short term if you can easily paste and un paste it at the top of a document.

**Comment by prater-devin** [comment]
I mainly use it on web and occasionally on mobile. I've not tried it on
desktop yet, since the silverbullet I use is hosted on a server that our
whole team uses as a wiki.

**Comment by tristansokol** [comment]
Does the markdown:preview command help with reading the page?

**Comment by prater-devin** [comment]
No, sorry, it doesn’t change the text area that I’ve noticed. I can read the page with the screen reader cursor, but when moving around in the text area, screen readers have the issue of speech stopping after formatted text, like links and such.
